src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestV4ServiceImpl.java

```diff
@@ -0,0 +1,344 @@
+        if (!ConcurUtils.validateFormatAndPrefixOfParameterizedUrl(queryUrl, getRequestV4Endpoint())) {
+            LOG.error("verifyQueryUrlForNextResultsPageIsValid, Detected an unexpected or malformed search URL: "
+                    + queryUrl);
+            throw new RuntimeException("Query URL for next page of search results is malformed");
+        }
+    }
+
+    protected ConcurRequestV4ListingDTO getPendingTravelRequests(String accessToken, int page, String queryUrl) {
+        String messageDetail = MessageFormat.format(
+                configurationService.getPropertyValueAsString(ConcurKeyConstants.MESSAGE_CONCUR_REQUESTV4_LISTING),
+                page);
+        return concurEventNotificationV2WebserviceService.buildConcurDTOFromEndpoint(
+                accessToken, queryUrl, ConcurRequestV4ListingDTO.class, messageDetail);
+    }
+
+    protected Stream<ConcurEventNotificationProcessingResultsDTO> processTravelRequestsSubset(
+            String accessToken, Map<String, String> testUserIdMappings, ConcurRequestV4ListingDTO requestListing) {
+        if (CollectionUtils.isEmpty(requestListing.getListItems())) {
+            return Stream.empty();
+        }
+        Stream.Builder<ConcurEventNotificationProcessingResultsDTO> subResults = Stream.builder();
+        
+        for (ConcurRequestV4ListItemDTO requestAsListItem : requestListing.getListItems()) {
+            if (StringUtils.isBlank(requestAsListItem.getId())) {
+                throw new IllegalStateException("Found a request item with a blank ID; this should NEVER happen");
+            }
+            if (isRequestPendingExternalValidation(requestAsListItem)
+                    && requestHasAppropriateOwner(requestAsListItem, testUserIdMappings)) {
+                ConcurEventNotificationProcessingResultsDTO processingResultForRequest = processTravelRequest(
+                        accessToken, requestAsListItem);
+                subResults.add(processingResultForRequest);
+            }
+        }
+        
+        return subResults.build();
+    }
+
+    protected boolean isRequestPendingExternalValidation(ConcurRequestV4ListItemDTO requestAsListItem) {
+        ConcurRequestV4StatusDTO approvalStatus = requestAsListItem.getApprovalStatus();
+        return ObjectUtils.isNotNull(approvalStatus) && StringUtils.equalsIgnoreCase(
+                RequestV4Status.PENDING_EXTERNAL_VALIDATION.name, approvalStatus.getName());
+    }
+
+    protected boolean requestHasAppropriateOwner(ConcurRequestV4ListItemDTO requestAsListItem,
+            Map<String, String> testUserIdMappings) {
+        ConcurRequestV4PersonDTO owner = requestAsListItem.getOwner();
+        if (ObjectUtils.isNull(owner) || StringUtils.isBlank(owner.getId())) {
+            return false;
+        }
+        boolean isTestUser = testUserIdMappings.containsKey(owner.getId());
+        return isProduction() != isTestUser;
```

Comment:
I think this logic is correct, but it is a bit confusing.  Can it be refactored?
When in production, only allow users that aren't test users
when in non production, only allow users that are test users.

Reply:
I'll update this to use the conditional/ternary operator or something similar; that should help make the intent clearer.